[rpc] add default arg for init_method

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30208 [rpc] add default arg for init_method**

Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

rohan-varma committed:
[wip][rpc] add default arg for init_method

Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct. Also fixes some docs.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip][rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct. Also fixes some docs.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip][rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct. Also fixes some docs.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip][rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct. Also fixes some docs.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

xush6528 commented on torch/csrc/distributed/rpc/rpc_agent.h:
```diff
@@ -15,6 +15,7 @@ namespace rpc {
 struct RpcBackendOptions {
   RpcBackendOptions() noexcept = default;
```
There is an relevant compiling error.

Try removing `noexcept` ? 

I guess it's a mismatch with std::string's default constructor.

mrshenli commented on torch/distributed/rpc/__init__.py:
```diff
@@ -50,15 +49,24 @@ def init_rpc(
                         128 characters.
-            init_method(str): backend specific init arguments.
+            init_method(str): URL specifying how to initialize the
+                                     RPC backend. Default is "env://" if no
```
indent here is wrong?

rohan-varma replied:
This is removed in the latest update.

mrshenli commented on torch/distributed/rpc/__init__.py:
```diff
@@ -50,15 +49,24 @@ def init_rpc(
+                                     ``init_method`` is specified.
             rank (int): a globally unique id/rank of this node.
             world_size (int): The number of workers in the group.
             rpc_backend_options (RpcBackendOptions): The options passed to RpcAgent
```
(This can come in a followup PR) Do we have docs for rpc_backend_options to show how to construct it, probably with a customized init_method?

rohan-varma replied:
I'll work on this in a follow up PR.

xush6528 commented on torch/distributed/rpc/__init__.py:
```diff
@@ -50,15 +49,24 @@ def init_rpc(
         """
+
+        if not rpc_backend_options:
+            # default construct a set of RPC agent options.
```
nit:  `# default construct a `RpcBackendOptions`.`

pritamdamania87 commented on torch/distributed/rpc/__init__.py:
```diff
@@ -50,15 +49,24 @@ def init_rpc(
                         underscore, and/or dash, and must be shorter than
                         128 characters.
-            init_method(str): backend specific init arguments.
+            init_method(str): URL specifying how to initialize the
```
Why do we still have this in the docs? Isn't this parameter removed from `init_rpc` now?

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct. Also fixes some docs.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

mrshenli reviewed (approved):
This stack looks good to go. Test failures are irrelevant:

```
Nov 22 10:11:16 ======================================================================
Nov 22 10:11:16 FAIL [0.024s]: test_pytorch_graph (__main__.TestTensorBoardPytorchGraph)
Nov 22 10:11:16 ----------------------------------------------------------------------
Nov 22 10:11:16 Traceback (most recent call last):
Nov 22 10:11:16   File "test_tensorboard.py", line 507, in test_pytorch_graph
Nov 22 10:11:16     self.assertTrue(compare_proto(graphdef, self))
Nov 22 10:11:16 AssertionError: False is not true
Nov 22 10:11:16 
Nov 22 10:11:16 ----------------------------------------------------------------------
```

xush6528 reviewed (approved):
Thanks

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add default arg for init_method"


Adds default arg for init_method so users don't have to pass this in,
and moves it to `RpcBackendOptions` struct.

Differential Revision: [D18630074](https://our.internmc.facebook.com/intern/diff/D18630074/)

[ghstack-poisoned]
